runatlantis.io/docs/server-configuration.md

```diff
@@ -607,6 +607,14 @@ and set `--autoplan-modules` to `false`.
   Hide previous plan comments to declutter PRs. This is only supported in
   GitHub currently. This is not enabled by default.
 
+### `--delete-prev-command-comments`
+  ```bash
+  atlantis server --delete-prev-command-comments
+  # or
+  ATLANTIS_DELETE_PREV_COMMAND_COMMENTS
+  ```
+  Deletes previous command comments to declutter the merge request. This is only supported in Gitlab currently. Github users should use `--hide-prev-plan-comments`. 
```

Comment:
Maybe we should add this functionality under `hide-prev-plan-comments` flag? since this is basically the same thing but for gitlab.

Reply:
@nitrocode

Reply:
@jamengual I had had initially put it there, but over thought it and didn't want to hide a Delete command behind a Hide flag was my reasoning. :)

Reply:
Yes i think we should keep consistency with how we have it with github. It would be better to hide than to delete in order to keep auditability.

Reply:
Yeah I wish Gitlab supported hiding, however, there is a Discussion feature on gitlab that supports user minimization.

I was toying around with the idea of for every command sent by the user turn that merge request comment to a discussion and then have atlantis reply to that discussion. 

This idea was born when tons of big plans/applys would slow the UI down significantly. 

I agree with the auditability of it.  I was also considering keeping the last 3 or remove the last plans on successful apply. 

I'd love to hear thoughts and feedback for sure! I appreciate the comments up to this point.

Reply:
If there's no way to hide, it would be even better to edit an existing comment if one exists. Could you implement that instead?

In github, every edit is auditable. If it is the same in gitlab, that may be even better and then that would use a new flag.

Reply:
The most ideal method is to use the gitlab equivalent of githubs checks api and that would completely avoid comments completely. I'm unsure if gitlab has this feature and i would not be surprised if they do.

Here is the equivalent checks api ticket we have for github https://github.com/runatlantis/atlantis/issues/936

Reply:
Oh i like this idea! Let me investigate it Id be happy to implement if its possible! I'll look into this thanks for the discussion @nitrocode

Reply:
Or how about update old plan with detail tag to hide it  and create new comment?

Reply:
@tyler-allen-tango friendly ping

Reply:
Slating some time today to look and experiment with this. I'll update when I know more. @nitrocode Would you want to close this MR until i can try to test with some of the recommendations above such as details tag on the html markup/attempting to move the outputs to a diff location in gitlab. Will probably take a little bit.

Reply:
Completely up to you. I do not mind (1) keeping this open, (2) re-using this one with the recommendations above, or (3) closing and having a separate PR to solve the same problem.